src/shogun/features/CombinedDotFeatures.cpp

```diff
@@ -114,7 +114,8 @@ float64_t CCombinedDotFeatures::dot(int32_t vec_idx1, CDotFeatures* df, int32_t
 	return result;
 }
 
-float64_t CCombinedDotFeatures::dense_dot(int32_t vec_idx1, const float64_t* vec2, int32_t vec2_len) const
+float64_t CCombinedDotFeatures::dot(
+    int32_t vec_idx1, const SGVector<float64_t> vec2) const
```

Comment:
The second argument should be a const ref rather than passing by value imo

Reply:
@karlnapf Doing stuff like `dot(index, vector.slice(a, b))` wouldn't work if we used an lvalue reference.

Reply:
I'm wrong, nvm :")